### PR TITLE
13507: FHIR Converter integration tests

### PR DESCRIPTION
**Pull request by mkalish**
This PR applies a fix for which message model is chosen when parsing an HL7 message by deferring to the topic as well as adding comprehensive integration tests for the FHIR converter

Test Steps:
1. Send one of the elims messages through the pipeline and verify that it gets delivered
2. Send one of the MarsOtc messages use to test validation through the pipeline and verify it gets delivered

Verify the coverage (generated from only running the new tests)
Screenshot 2024-04-26 at 2 49 48 PM

## Changes
- Adds the ability to use an HL7MessageParseConfiguration set on the topic (required to support the RadXMars spec), otherwise uses the current fallback behavior of 
  - Any ORU_R01 message is parsed as 27 and otherwise the default HAPI behavior is use
- Adds a comprehensive set of integration tests for the FHIR Converter

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #13507

### CODE REVIEW

**Comment by david-navapbc** [comment]
in addition to the integration test failures - there is an additional test failure that only presents when `gradlew package` or `gradlew build` is invoked. When `gradlew test` is invoked the test does not fail. 

```gov.cdc.prime.router.fhirengine.translation.SyncSchemaE2ETests test end-to-end sync schemas workflow() FAILED (10s)

  gov.cdc.prime.router.fhirengine.translation.TranslationSchemaManager$Companion$TranslationSyncException: Validation state was invalid, the valid blob is misconfigured. 
  It is either duplicated or not present when the state is not being validated
      at app//gov.cdc.prime.router.fhirengine.translation.SyncSchemaE2ETests.test end-to-end sync schemas workflow(SyncSchemaE2ETests.kt:87)```

File: prime-router/settings/organizations.yml
Comment (mkalish) [inline prime-router/settings/organizations.yml:109]: A little unrelated, but I realized we should have a mars otc sender for test purposes

File: prime-router/src/main/kotlin/Sender.kt
Comment (mkalish) [inline prime-router/src/main/kotlin/Sender.kt:46]: I had initially missed that this needs to get added in addition to topic

File: prime-router/src/main/resources/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml
Comment (mkalish) [inline prime-router/src/main/resources/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml:591]: I originally encountered this when trying to write tests for an HL7 message with a sender transform, but I worked around it by using the original-pipeline-transform (there were othe rreasons that it made impossible to use), but I didn't want to lose track of this and will open a bug before merging.

File: prime-router/src/test/kotlin/fhirengine/azure/FHIRConverterIntegrationTests.kt
Comment (mkalish) [inline prime-router/src/test/kotlin/fhirengine/azure/FHIRConverterIntegrationTests.kt:155]: I'll add a ticket for this as well, so both formats don't need to be passed

**Comment by adegolier** [comment]
Haven't been able to do the test steps yet because I'm having problems building/running the tests locally.

File: prime-router/src/test/kotlin/common/TestcontainersUtils.kt
Comment (mkalish) [inline prime-router/src/test/kotlin/common/TestcontainersUtils.kt:18]: The caching here was not safe across multiple gradle test runners and can result in random tests beginning to fail even if they were previously passing.

File: prime-router/src/main/kotlin/SettingsProvider.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/SettingsProvider.kt:57]: Why wasn't MARS_OTC_ELR updated to configure `HL7MessageParseAndConvertConfiguration`?

File: prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt:278]: I think we need to look into `getHL7ParsingContext` and evaluate the business logic there and align with product. I'll think on it and maybe create a ticket, but I think this is fine for now if its well tested.

File: prime-router/src/test/kotlin/fhirengine/azure/FHIRConverterIntegrationTests.kt
Comment (arnejduranovic) [reply prime-router/src/test/kotlin/fhirengine/azure/FHIRConverterIntegrationTests.kt:155]: reminder to create ticket and then update this comment with the number

File: prime-router/src/main/kotlin/SettingsProvider.kt
Comment (mkalish) [reply prime-router/src/main/kotlin/SettingsProvider.kt:57]: I was holding off on adding that until the mappings and java file is done.

File: prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt
Comment (mkalish) [reply prime-router/src/main/kotlin/fhirengine/utils/HL7Reader.kt:278]: Yep... I think we have it isolated well now so that we make tactical changes here

**Review by arnejduranovic — APPROVED** [review]
Changes look good and tests pass locally

File: prime-router/src/main/kotlin/SettingsProvider.kt
Comment (mkalish) [reply prime-router/src/main/kotlin/SettingsProvider.kt:57]: FYI, my thought is that the fast follow is done in [this](https://github.com/CDCgov/prime-reportstream/pull/14255) PR, let me know if you have a different preference

**Pull request merged**
